azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -617,4 +622,41 @@ public static void deleteSymlinkedFile(final Path symlinkedFilePath)
       throw new ExecutorManagerException(e);
     }
   }
+
+  /**
+   * Shutdown the Container. This shuts down the ExecutorService which runs the flow execution
+   * as well as JettyServer.
+   */
+  private void shutdown() {
+    logger.info("Shutting down the pod");
+    while (!this.flowFuture.isDone()) {
+      // This should not happen immediately as submitFlowRunner is a blocking call.
+      try {
+        Thread.sleep(100);
+      } catch (final InterruptedException e) {
+        logger.error("The sleep while waiting for execution : {} to finish was interrupted",
+                this.flowRunner.getExecutionId());
+      }
+    }
+
+    try {
+      this.executorService.shutdown();
+      boolean result = false;
+      while (!result) {
+        logger.info("Awaiting Shutdown of executing flow");
+        try {
+          result = this.executorService.awaitTermination(5, TimeUnit.SECONDS);
+        } catch (final InterruptedException e) {
+          logger.error(e.getMessage());
+        }
+      }
+      this.jettyServer.stop();
+      this.jettyServer.destroy();
+    } catch (final Exception e) {
+      // Eat up the exception
+      logger.error("Error shutting down JettyServer while winding down the FlowContainer", e);
```

Comment:
The `try{}` block is not specific to JettyServer, please either make the message more generic or alternatively break it into separate `try...catch` blocks (e.g for jettrserver shutdown, executorservice shutdown etc)

Reply:
Good call. Earlier shutdown was out of the block. Let me take care of this.